### PR TITLE
feat: add `react-server` export condition

### PR DESCRIPTION
**Pull request by stipsan**
This improves compatibility with `app-router`. This is not a breaking change as `app-router` doesn't work correctly in production atm. We don't have full compatibility yet (Next.js app-router uploads, Server Actions, custom headers maybe) but it's a step in the right direction.

This means `fetch: true` is no longer necessary to opt-in for automatic request deduping and using `fetch` under the hood.
Why is this good?
Because if `fetch: true` is used in Next.js pages-router, or code that runs outside app-router, then it's probably not what you want. It supports less features, and isn't designed to run in that environment.

This PR doesn't remove `fetch: true` though, as it's still useful in other scenarios, like `Deno` and similar edge cases.

### CODE REVIEW

File: test-next/next.config.mjs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
+  transpilePackages: ['get-it'],
```
Comment (stipsan) [inline test-next/next.config.mjs:3]: This is only necessary because we're on a monorepo. Next.js transpiles by default unless you [opt-out](https://nextjs.org/docs/app/api-reference/next-config-js/serverComponentsExternalPackages).
For some reason when it's using a symlink it skips transpiling when using `next build`, however `next dev` transpiles 🤷
